### PR TITLE
Update prep metrics to add samtools stats in rnaaln

### PR DESCRIPTION
**Pull request by guanqiaofeng**
In the previous `rnaaln` workflow, the `QC metric` did not include `samtools_stats`. We have decided to add `samtools_stats` to the `QC metric`. The `prep_metrics` module has been updated to handle `samtools_stats` for `rnaaln`.

### CODE REVIEW

File: modules/icgc-argo-workflows/prep/metrics/resources/usr/bin/dnaaln.py [hunk 1]
```diff
@@ -47,7 +47,7 @@
      'total_pf_reads': 'sequences',
      'average_base_quality': 'average_quality',
      'average_read_length': 'average_length',
-     'reads_duplicated_percent': 'reads_duplicated_percent',
+     'pct_reads_duplicated': 'reads_duplicated_percent',
```
Comment (guanqiaofeng) [inline modules/icgc-argo-workflows/prep/metrics/resources/usr/bin/dnaaln.py:50]: We handle `percent` as `pct` for standardization. This one was missed. Now correct it to the standard format

File: modules/icgc-argo-workflows/prep/metrics/resources/usr/bin/rnaaln.py [hunk 2]
```diff
@@ -29,17 +29,15 @@
 tool_fieldmap = { # 'target name' : 'original name'
     'hisat2': {
         'paired_total' : 'paired_total',
-        'unpaired_total' : 'unpaired_total',
-        'overall_alignment_rate' : 'overall_alignment_rate'
```
Comment (guanqiaofeng) [inline modules/icgc-argo-workflows/prep/metrics/resources/usr/bin/rnaaln.py:33]: `overall_alignment_rate` in `hisat2` is duplicated to `pct_reads_mapped` in `samtools_stats`. So removing it

File: modules/icgc-argo-workflows/prep/metrics/resources/usr/bin/rnaaln.py [hunk 3]
```diff
@@ -29,17 +29,15 @@
...
+        'unpaired_total' : 'unpaired_total'
```
Comment (guanqiaofeng) [inline modules/icgc-argo-workflows/prep/metrics/resources/usr/bin/rnaaln.py:32]: `overall_alignment_rate` in `hisat2` is duplicated to `pct_reads_mapped` in `samtools_stats`. So removing it

File: modules/icgc-argo-workflows/prep/metrics/resources/usr/bin/rnaaln.py [hunk 4]
```diff
@@ -29,17 +29,15 @@
...
     },
     'star' : {
         'total_reads' : 'total_reads',
         'avg_input_read_length' : 'avg_input_read_length',
-        'uniquely_mapped_percent' : 'uniquely_mapped_percent',
+        'pct_uniquely_mapped' : 'uniquely_mapped_percent',
```
Comment (guanqiaofeng) [inline modules/icgc-argo-workflows/prep/metrics/resources/usr/bin/rnaaln.py:37]: We handle `percent` as `pct` for standardization. This one was missed. Now correct it to the standard format

File: modules/icgc-argo-workflows/prep/metrics/resources/usr/bin/rnaaln.py [hunk 5]
```diff
@@ -29,17 +29,15 @@
...
         'avg_mapped_read_length' : 'avg_mapped_read_length',
         'num_splices' : 'num_splices',
         'num_annotated_splices' : 'num_annotated_splices',
-        'mismatch_rate' : 'mismatch_rate',
```
Comment (guanqiaofeng) [inline modules/icgc-argo-workflows/prep/metrics/resources/usr/bin/rnaaln.py:42]: `mismatch_rate` in `star` is duplicated to `mismatch_bases_rate` in `samtools_stats`. So removing it.

File: modules/icgc-argo-workflows/prep/metrics/resources/usr/bin/rnaaln.py [hunk 6]
```diff
@@ -54,6 +52,19 @@
...
+        'mean_insert_size': 'insert_size_average',
+        'insert_size_std_deviation': 'insert_size_standard_deviation',
+        'total_pf_reads': 'sequences',
+        'average_base_quality': 'average_quality',
+        'average_read_length': 'average_length',
+        'pct_reads_duplicated': 'reads_duplicated_percent',
+        'non-primary_alignments': 'non-primary_alignments',
+        'pairs_on_different_chromosomes': 'pairs_on_different_chromosomes',
+        'mismatch_bases_rate': 'error_rate'
     }
```
Comment (guanqiaofeng) [inline modules/icgc-argo-workflows/prep/metrics/resources/usr/bin/rnaaln.py:68]: This section extract the same fields in `samtools_stats` as `dnaaln.py`

**Review by lindaxiang — APPROVED** [review]
LGTM

**Review by edsu7 — CHANGES_REQUESTED** [review]

File: tests/data/qa/multiqc_data/rnaaln/star/multiqc_sources.txt [hunk 7]
```diff
@@ -1,3 +1,4 @@
 Module	Section	Sample Name	Source
-Picard	RnaSeqMetrics	SA622799	/Users/gfeng/Documents/Work/ICGC-ARGO-WORKFLOWS/rnaaln/update_prep_metrics/nf-core-rnaaln/work/60/aa5f1eba43871dd9572177be612f40/3/TCRB-CA.DO262498.SA622799.csort.markdup.star.rna_metrics
-STAR	SummaryLog	SA622799	/Users/gfeng/Documents/Work/ICGC-ARGO-WORKFLOWS/rnaaln/update_prep_metrics/nf-core-rnaaln/work/60/aa5f1eba43871dd9572177be612f40/2/TCRB-CA.DO262498.SA622799.S01L1.Log.final.out
+Picard	RnaSeqMetrics	SA622799	/Users/gfeng/Documents/Work/ICGC-ARGO-WORKFLOWS/rnaaln/compile/nf-core-rnaaln/work/97/cc05f1053a0029a3b30559797f7f62/4/TCRB-CA.DO262498.SA622799.csort.markdup.star.rna_metrics
```
[the remark this answers is not in the document]
Comment (guanqiaofeng) [reply tests/data/qa/multiqc_data/rnaaln/star/multiqc_sources.txt:2]: Other test files also include personal directory. This file is generated by multiqc step. Maybe we can leave it as it is?

**Pull request merged**